### PR TITLE
Auto mod blame wolf

### PR DESCRIPTION
**Pull request by Skillz4Killz**
Adds the Automod feature

### CODE REVIEW

File: types/shared.ts [hunk 1]
```diff
@@ -215,6 +215,8 @@ export enum GuildFeatures {
   PrivateThreads = "PRIVATE_THREADS",
   /** Guild is able to set role icons */
   RoleIcons = "ROLE_ICONS",
+  /** Guild has set up auto moderation rules */
+  AutoModeration = "ANIMATED_BANNER",
```
Comment (meister03) [inline types/shared.ts:219]: What does this has to do with AutoModeration? type?

Comment (meister03) [inline types/shared.ts:219]:
```suggestion
  AutoModeration = "AUTO_MODERATION",
```

**Review by meister03 — CHANGES_REQUESTED** [review]
Looks  good will try to test it today

File: types/shared.ts (hunk 1 above)
Comment (Skillz4Killz) [reply types/shared.ts]: Copy paste typo

**Review by meister03 — DISMISSED** [review]

File: bot.ts [hunk 2]
```diff
@@ -484,6 +496,10 @@ export type RestManager = ReturnType<typeof createRestManager>;
 
 export interface EventHandlers {
   debug: (text: string, ...args: any[]) => unknown;
+  automodRuleCreate: (bot: Bot, rule: AutoModerationRule) => unknown;
```
Comment (nsylke) [inline bot.ts:499]:
```suggestion
  automodRuleCreate: (bot: Bot, rule: AutoModerationRule) => void;
```

File: bot.ts [hunk 3]
```diff
@@ -484,6 +496,10 @@ export type RestManager = ReturnType<typeof createRestManager>;
...
+  automodRuleUpdate: (bot: Bot, rule: AutoModerationRule) => unknown;
```
Comment (nsylke) [inline bot.ts:500]:
```suggestion
  automodRuleUpdate: (bot: Bot, rule: AutoModerationRule) => void;
```

**Review by nsylke — CHANGES_REQUESTED** [review]
See #2285 for reason

**Comment by lts20050703** [comment]
See https://github.com/discordeno/discordeno/pull/2285#issuecomment-1154434308 for the reason to Resolve all conversation regarding suggested changes

**Comment by lts20050703** [comment]
@Skillz4Killz Check Formatting failed. Please run `deno fmt`

**Pull request merged**